### DISCUSSION TITLE
WebRTC - Set usage limit

### DISCUSSION THREAD

**Discussion by sergiobellini** [bluenviron/mediamtx#3521] [Questions]
### Question

Hi everyone, I have a MediaMTX server with only the WebRTC service active, connected to a website with iframe (and SSL certificates).
To avoid overloading the server where MediaMTX runs, is it possible to set a timeout? That is, after 5 minutes of viewing, the video automatically ends?

**Comment by aler9** [comment]
Hello, this logic can be easily implemented by writing a script that:

1. connects to MediaMTX API and downloads list of WebRTC readers through `/v3/webrtcsessions/list`
2. filters WebRTC readers with `state = read` and `created < (now - 5 minutes)`
3. kicks these readers through `/v3/webrtcsessions/kick`

**Reply by sergiohacs** [reply]
I really hadn't thought about APIs, thank you very much!

[a comment by an automated account is not shown]
